### PR TITLE
expose more host functions

### PR DESCRIPTION
**Pull request by flavio**
This PR performs the following changes:

- Add dns lookup API: this is needed to solve [this feature request](https://github.com/kubewarden/policy-server/issues/236)
- Expose oci get manifest API

### CODE REVIEW

File: src/host_capabilities/net.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+/// A list of IP addresses
+pub type LookupResponse = Vec<String>;
+
+/// Lookup the addresses for a given hostname via DNS
+pub fn lookup_host(host: &str) -> Result<LookupResponse> {
+    let req = json!(host);
+    let msg = serde_json::to_vec(&req)
+        .map_err(|e| anyhow!("error serializing the validation request: {}", e))?;
+    let response_raw = wapc_guest::host_call("kubewarden", "net", "dns_lookup_host", &msg)
```
Comment (raulcabello) [inline src/host_capabilities/net.rs:12]: Do we want to add versioning as we are doing for [verification](https://github.com/kubewarden/policy-sdk-rust/blob/main/src/host_capabilities/verification.rs#L68)? Unlike verification I don't expect `dns_lookup_host` or `oci_manifest` to change, so I am fine to leave it without versioning.

File: src/host_capabilities/oci.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+use anyhow::{anyhow, Result};
+use serde_json::json;
+
+/// Computes the digest of the OCI object referenced by `image`
+pub fn manifest_digest(image: &str) -> Result<String> {
+    let req = json!(image);
+    let msg = serde_json::to_vec(&req)
+        .map_err(|e| anyhow!("error serializing the validation request: {}", e))?;
+    let response_raw = wapc_guest::host_call("kubewarden", "oci", "manifest_digest", &msg)
```
Comment (raulcabello) [inline src/host_capabilities/oci.rs:9]: same here. do we want versioning?

**Review by raulcabello — APPROVED** [review]
LGTM, thanks! just a comment regarding if we should add versioning or not.

File: src/host_capabilities/net.rs (hunk 1 above)
Comment (flavio) [reply src/host_capabilities/net.rs:12]: yeah, I don't think we need versioning but honestly I don't think it should harm. I'll version them

**Review by ereslibre — APPROVED** [review]
LGTM besides @raulcabello's comments. I agree with versioning them.

**Review by jvanz — APPROVED** [review]
Besides the comments from others, LGTM. Thank you!

File: src/host_capabilities/net.rs (hunk 1 above)
Comment (flavio) [reply src/host_capabilities/net.rs]: done

File: src/host_capabilities/oci.rs (hunk 2 above)
Comment (flavio) [reply src/host_capabilities/oci.rs]: done

**Review by raulcabello — APPROVED** [review]
LGTM, thanks!

**Review by viccuad — APPROVED** [review]
LGTM!

**Comment by flavio** [comment]
Alright, this is ready to be reviewed one last time. I've changed the response types to be real objects. This is needed to make TinyGo work with easyjson, as discussed during our daily

**Review by raulcabello — APPROVED** [review]
LGTM

**Pull request merged**